src/Hosting/Abstractions/src/IStartup.cs

```diff
@@ -7,10 +7,21 @@
 
 namespace Microsoft.AspNetCore.Hosting
 {
+    /// <summary>
+    /// Provides an interface for initializing services and middlewares used by an application.
+    /// /// </summary>
```

Comment:
This is a bit messed up. @Tratcher is there a conceptual doc we can use to expand this description?